spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configserver/src/main/java/org/springframework/cloud/kubernetes/configserver/KubernetesEnvironmentRepository.java

```diff
@@ -61,18 +63,25 @@ public Environment findOne(String application, String profile, String label, boo
 		LOG.info("Application: " + application);
 		LOG.info("Label: " + label);
 		Environment environment = new Environment(application, profiles, label, null, null);
-		try {
-			StandardEnvironment springEnv = new StandardEnvironment();
-			springEnv.setActiveProfiles(profiles);
-			if (!"application".equalsIgnoreCase(application)) {
-				addApplicationConfiguration(environment, springEnv, application);
-			}
-			addApplicationConfiguration(environment, springEnv, "application");
-			return environment;
+		ArrayUtils.reverse(profiles);
+		for (String activeProfile : profiles) {
```

Comment:
Before this change the KubernetesPropertySourceSuppliers would return a single PropertySource with all the properties aggregated together.  This makes sense when the app is using the K8S API to directly fetch its configuration.  However when the app is leveraging the Kuberentes Config Server we actually want to return a separate PropertySource for not only each configmap/secret we find but also each data element inside the config map/secret. 

So instead of creating one `StandardEnvironment` with all the requested profiles and getting back a single `PropertySource` we create a `StandardEnvironment` for each profile so we add a `PropertySource` for each one.

In addition we were not also fetching the `default` profile in the case where a profile is requested from the config server so I have added some logic to make sure we return that data as well.

Reply:
Correct.  See the example from the config server in the PR description that I added.  When you request `k8s,prod` the configuration for the `default` profile is also returned

Reply:
All of what you said is 100% correct.

The problem with the was the code works today at least for the config server is that the property sources for `account-k8s` and `account-prod` would include the properties from `account` as well and that is not technically correct because those property sources don't actually contain those properties.

This is why I needed to make changes here
https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1600/files#diff-7fd0d91605f84c420115afd2e5ace4fa142b14471c72c82499c36b17cdc4fb17R208

and here

https://github.com/spring-cloud/spring-cloud-kubernetes/pull/1600/files#diff-1f76a87de9a51d26c13cb50cfd60090c103ac4c69f36d7e85481654251f75c62R109

In the case where the we are looking at config map or secret that has an active profile appended to it we should not include any properties from the default profile (because we will have a separate property source for that).